src/ReflectionClosure.php

```diff
@@ -349,13 +361,17 @@ public function getCode()
                         case T_STATIC:
                         case T_NS_SEPARATOR:
                         case T_STRING:
-                            $id_start = $token[1];
-                            $id_start_ci = strtolower($id_start);
-                            $id_name = '';
-                            $context = 'root';
-                            $state = 'id_name';
-                            $lastState = 'closure';
-                            break 2;
+                            if ($context !== 'string') {
```

Comment:
This is partially correct. But when we are inside a string we can also have `T_DOLLAR_OPEN_CURLY_BRACES`
```php
namespace Test;
const foo = 0;

$x = [0 => 'Constant foo', 'foo' => 'String foo'];

echo "$x[foo]\n"; // here foo is a string
echo "${x[foo]}\n"; // here foo is a constant
```

Reply:
Yeah where is the problem, I just test it and its work without any problem
I think the problem was here: https://github.com/opis/closure/commit/85fec0b113e163c343c9d643744bc1221641f0b9

Reply:
It doesn't work

```php
namespace Test;
const foo = 1;

$f = function ($x) {
  return "${x[foo]}";
};

echo \Opis\Closure\serialize($f); // body should contain: return "${x[\Test\foo]}";

```

Reply:
before this commit: https://github.com/opis/closure/commit/85fec0b113e163c343c9d643744bc1221641f0b9
```php
$a = [0 => "bar"];
$c = function() use($a)
{
    return "${a[test_array_key_for_bar_value]}";
};
```
Test result:
> Failed asserting that two strings are equal.
> Expected :'{a[test_array_key_for_bar_value]}'
> Actual   :'bar'
- after this commit: https://github.com/opis/closure/commit/85fec0b113e163c343c9d643744bc1221641f0b9
> Test result:
> OK (1 test, 1 assertion)

Reply:
Hmmm, so body should contain: return "${x[\Test\foo]}"
I will check now

Reply:
@abdrzakoxa Please read my example. You should notice some things:

a `namespace`, a `const` declaration and NO `define()` used

```php
namespace Test;
const foo = 1;
```

From another file you can `use` the const

```php
use const Test\foo as aliased_foo;
```

Reply:
@sorinsarca Solved now